presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestCustomJsonDateTimeFormatter.java

```diff
@@ -0,0 +1,102 @@
+import static io.prestosql.testing.DateTimeTestingUtils.sqlTimestampOf;
+import static io.prestosql.testing.DateTimeTestingUtils.sqlTimestampWithTimeZoneOf;
+import static io.prestosql.testing.assertions.Assert.assertEquals;
+
+public class TestCustomJsonDateTimeFormatter
+{
+    private static final DateTimeFormat DATE_TIME_FORMAT = DateTimeFormat.CUSTOM_DATE_TIME;
+
+    private void testDate(SqlDate value, String formatHint, String expectedLiteral)
+    {
+        String actualLiteral = DATE_TIME_FORMAT.getFormatter(Optional.of(formatHint)).formatDate(value);
+        assertEquals(actualLiteral, expectedLiteral);
+    }
+
+    private void testTime(SqlTime value, String formatHint, int precision, String expectedLiteral)
+    {
+        String actualLiteral = DATE_TIME_FORMAT.getFormatter(Optional.of(formatHint)).formatTime(value, precision);
+        assertEquals(actualLiteral, expectedLiteral);
+    }
+
+    private void testTimeWithTZ(SqlTimeWithTimeZone value, String formatHint, String expectedLiteral)
+    {
+        String actualLiteral = DATE_TIME_FORMAT.getFormatter(Optional.of(formatHint)).formatTimeWithZone(value);
+        assertEquals(actualLiteral, expectedLiteral);
+    }
+
+    private void testTimestamp(SqlTimestamp value, String formatHint, String expectedLiteral)
+    {
+        String actualLiteral = DATE_TIME_FORMAT.getFormatter(Optional.of(formatHint)).formatTimestamp(value);
+        assertEquals(actualLiteral, expectedLiteral);
+    }
+
+    private void testTimestampWithTZ(SqlTimestampWithTimeZone value, String formatHint, String expectedLiteral)
+    {
+        String actualLiteral = DATE_TIME_FORMAT.getFormatter(Optional.of(formatHint)).formatTimestampWithZone(value);
+        assertEquals(actualLiteral, expectedLiteral);
+    }
+
+    @Test
+    public void testCustomDateTimeFunctions()
+    {
+        testDate(sqlDateOf(2020, 8, 14), "yyyy-MM-dd", "2020-08-14");
+        testDate(sqlDateOf(1970, 1, 1), "yyyy-MM-dd", "1970-01-01");
+        testDate(sqlDateOf(1900, 1, 1), "yyyy-MM-dd", "1900-01-01");
+        testDate(sqlDateOf(3001, 1, 1), "yyyy-MM-dd", "3001-01-01");
+
+        testTime(sqlTimeOf(3, 15, 36, 25, 123000000), "HH:mm:ss.SSS", 3, "15:36:25.123");
+        testTime(sqlTimeOf(3, 15, 36, 25, 0), "HH:mm:ss", 3, "15:36:25");
+        testTime(sqlTimeOf(3, 8, 12, 45, 987000000), "HH:mm:ss.SSS", 3, "08:12:45.987");
+        testTime(sqlTimeOf(3, 0, 0, 0, 0), "HH:mm:ss.SSS", 3, "00:00:00.000");
+        testTime(sqlTimeOf(3, 23, 59, 59, 999000000), "HH:mm:ss.SSS", 3, "23:59:59.999");
+
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(10, 23, 35, 123000000, ZoneOffset.UTC), "HH:mm:ss.SSS Z", "10:23:35.123 +0000");
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(0, 0, 0, 0, ZoneOffset.UTC), "HH:mm:ss.SSS Z", "00:00:00.000 +0000");
+        testTimeWithTZ(sqlTimeWithTimeZoneOf(10, 23, 35, 123000000, ZoneOffset.ofHours(2)), "HH:mm:ss.SSS Z", "10:23:35.123 +0000");
```

Comment:
Why zone in expected value is "+0000"?
Looks like `.withZoneUTC();` in `CustomDateTimeFormatter` is the culprit.
Is that a conscious decision that we do not want to expose the timezone information from the value in the encoded string?

Reply:
The decoder assumes UTC time zone, so that's what I figured the encoder should do as well. Is this assumption incorrect?